### PR TITLE
Fix cloning not adding traits

### PR DESCRIPTION
**Pull request by DEATHB4DEFEAT**
# About the PR

- Makes TraitSystem into something usable by other systems
- Uses TraitSystem in CloningSystem to add traits to the clones
- Cloning emits BeingClonedEvent for other systems to check if they should be getting cloned
- Cloning emits a BeenClonedEvent for other systems to do whatever to accepted clones
- Move AddTraits and AddComponentSpecial to their own EntitySystems, listening for the BeenCloned event.

# Changelog

:cl: DEATHB4DEFEAT
- fix: Fixed cloning not applying traits.

### CODE REVIEW

File: Content.Server/Traits/TraitSystem.cs [hunk 1]
```diff
@@ -15,25 +17,52 @@ public override void Initialize()
...
 
+    /// <summary>
+    ///     All the jobs that can't have traits.
+    /// </summary>
     private readonly List<string> _jobBlacklist = new()
     {
         "SAI",
         "Cyborg",
         "MedicalCyborg"
     };
```
Comment (Pspritechologist) [inline Content.Server/Traits/TraitSystem.cs]: Feel like this should either be a YAML, or a true/false in the YAML for the role or entity.

File: Content.Server/Traits/TraitSystem.cs [hunk 2]
```diff
@@ -45,35 +74,37 @@ private void OnPlayerSpawnComplete(PlayerSpawnCompleteEvent args)
...
             traits.Add(traitPrototype);
         }
 
         if (!addTraits)
         {
-            Logger.Warning($"Not adding traits to {args.Player.Name} because they broke the whitelist/blacklist of a negative trait.");
-            return;
+            Logger.Warning($"Not adding traits to entity {mob} because they broke the whitelist/blacklist of a negative trait.");
+            return false;
         }
```
Comment (Pspritechologist) [inline Content.Server/Traits/TraitSystem.cs]: This should give some player feedback, even if it's minor. Something pasted in chat or whatever could be good, maybe a message from [server]. Just so they have a way of knowing what happened.

**Review by Pspritechologist — CHANGES_REQUESTED** [review]
This seems like a really good change, and like it makes a lot of sense for a system that is frankly somewhat archaic, but do you worry it will be annoying to maintain at all? I feel this belongs on an Upstream PR before we merge it ourselves.

**Comment by DEATHB4DEFEAT** [comment]
> do you worry it will be annoying to maintain at all? I feel this belongs on an Upstream PR before we merge it ourselves.

I have accepted I will need to maintain the many changes I have made to the trait systems.
If wizden fixes the trait cloning, good for them, we can see how they did it and maybe modify my changes.

**Review by Pspritechologist — APPROVED** [review]

**Pull request merged**